src/useUrlQueryState.ts

```diff
@@ -13,7 +13,7 @@ export const useUrlQueryState = (name: string, overrideInitialValue?: string | n
       }
     }
     const paramValue = searchParams.get(name);
-    return paramValue === null ? undefined : paramValue;
+    return paramValue === null ? defaultValue : paramValue;
```

Comment:
can you try to make a `useDateUrlQueryState` below please just like the `useIntegerUrlQueryState` but for dates